src/components/Common/CustomAudioPlayer/index.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useRef } from "react";
+
+type Props = {
+  src: string;
+  muted: boolean;
+  play: boolean;
+  loop: boolean;
+  volume?: number;
+};
+
+const CustomAudioPlayer = ({
+  muted = false,
+  volume = 0.3,
+  src,
+  play,
+  loop,
+}: Props) => {
+  const audioElementRef = useRef<HTMLAudioElement>(new Audio(src));
+
+  useEffect(() => {
+    const audioElement = audioElementRef.current;
+
+    audioElement.src = src;
+    audioElement.loop = loop;
+    audioElement.muted = muted;
+    audioElement.volume = volume;
+
+    const controlAudio = async () => {
+      if (play) {
+        audioElement.currentTime = 0;
+        try {
+          await audioElement.play();
+        } catch (error) {
+          console.error("Audio play failed", error);
+        }
+      } else {
+        audioElement.pause();
+      }
```

Comment:
```suggestion
      if (play === false) {
        audioElement.pause();
        return;
      }
      
      audioElement.currentTime = 0;
      try {
        await audioElement.play();
      } catch (error) {
        console.error("Audio play failed", error);
      }
```
Early Return 패턴을 사용해도 좋을 것 같습니다.